### DISCUSSION TITLE
Missing interface description/alias via SNMP

### DISCUSSION THREAD

**Discussion by thannaske** [akvorado/akvorado#523] [Q&A] [answered]
Hey together, I'm not quite sure if that's worth a bug report or if I'm doing something wrong here. We're fetching the SNMP data for incoming flows from a Juniper MX204. I can see that Akvorado e.g. detects the interface's name and speed. However, the interface alias or description is missing, although that information is available via SNMP (confirmed with `snmpwalk`). That results in issues with the classification because we can't match against the interface description.

In this snippet you can see that the interface name and speed is populated but the description is not:
```
OutIfBoundary | internal
OutIfConnectivity |  
OutIfDescription |  
OutIfName | et-0/0/2.0
OutIfProvider |  
OutIfSpeed | 40000
```

The MX204 is configured as per akvorado documentation and the fact that e.g. the speed and name is displayed leads me to the conclusion that the SNMP connection per se seems to work.

Has anyone experienced a similar issue before or can give me a pointer on how to further debug that issue?

**Comment by vincentbernat** [comment, marked as answer]
The description needs to be set on the L3 interface (`unit 0` for example), not on the physical one. I think that's why you don't see it.

**Reply by thannaske** [reply]
That was indeed the solution to the issue, thank you! 

For the record, what I did was basically amending the configuration section of the interface:

```
thannaske@mx204 # show interfaces et-0/0/0
description "Transit: XXXXXXXX [n Gbps]";
unit 0 {
    description "Transit: XXXXXXX";
}
```

Newly added was the `unit 0` section as per your example. Once this was done and the SNMP cache was invalidated, everything was working fine.

**Reply by Lagertonne** [reply]
That's funny, I had exactly the opposite problem: I enabled sflow on the logical interface (i.e. xe-0/1/0.0) but the flows are sent from the MX204 with the SNMP ifIndex of the physical interface (xe-0/1/0), so I had to set the description on the physical, rather the logical one.

Did you take any extra steps to send the right SNMP ifIndex?

**Reply by deanp03** [reply]
Hi @Lagertonne , I have the same issue. From my testing, appears sflow and netflow pull through differently on akvorado. 

Looking at the SNMP data, the physical interface index is being used for sflow, but we need to capture logical interface descriptions. As sflows are sent from the MX204 with the SNMP ifIndex of the physical interface.

A user using _netflow_ works fine, example description on logical interface. junos-syntax: 
set interfaces xe-0/1/0 unit 0 description "Transit: Tier-1 IP TRANSIT"  <-- working on logical int netflow. 
set interfaces xe-0/1/0 description "Transit: Tier-1 IP TRANSIT"  <-- not working on PHY int netflow. 

When using _sflow_ the description on logical interfaces not working, example junos-syntax: 
set interfaces xe-0/1/0 unit 0 description "Transit: Tier-1 IP TRANSIT <-- not working on logical int sflow. 
set interfaces xe-0/1/0 description "Transit: Tier-1 IP TRANSIT <-- working on PHY int sflow. 

@vincentbernat any guidance on this one?

**Reply by vincentbernat** [reply]
Nothing special: either put the description on both IFD and IFL, or put it on the right one depending if you are using sFlow or NetFlow.
